prow/config/config.go

```diff
@@ -1403,11 +1405,13 @@ func ReadJobConfig(jobConfig string, yamlOpts ...yaml.JSONOpt) (JobConfig, error
 			return nil
 		}
 
-		base := filepath.Base(path)
-		if uniqueBasenames.Has(base) {
-			return fmt.Errorf("duplicated basename is not allowed: %s", base)
+		if requireUniqueBasenames {
```

Comment:
It is a tad weird that this check is implemented in the config loading itself and not in checkconfig..?